test/widgets/login_page_test.dart

```diff
@@ -0,0 +1,235 @@
+
+import '../util/mocks/mock_server.dart';
+import '../util/mocks/request_processor.dart';
+import '../util/mocks/request_processor.mocks.dart';
+import '../util/pump_material.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+  const String email = 'motismitfahrapp@gmail.com';
+  const String password = 'abc&EFG0';
+
+  setUpAll(() async {
+    MockServer.setProcessor(processor);
+    SupabaseManager.setCurrentProfile(null);
+  });
+
+  group('LoginPage', () {
+    final Finder emailFieldFinder = find.descendant(
+      of: find.byKey(const Key('loginEmailField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder passwordFieldFinder = find.descendant(
+      of: find.byKey(const Key('loginPasswordField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder loginButtonFinder = find.byType(LoadingButton);
+
+    testWidgets('Shows the login page', (WidgetTester tester) async {
+      await pumpMaterial(tester, const LoginPage());
+
+      final Finder loginFormFinder = find.byType(LoginForm);
+      final LoginFormState formState = tester.state(loginFormFinder);
+
+      expect(formState.buttonState, ButtonState.idle);
+    });
+
+    testWidgets('Validates the email', (WidgetTester tester) async {
+      await pumpMaterial(tester, const LoginPage());
+
+      final LoginFormState formState = tester.state(find.byType(LoginForm));
+
+      final FormFieldState emailField = tester.state(emailFieldFinder);
+
+      // Not validated yet, so no error
+      expect(emailField.hasError, isFalse);
+
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Empty, so error
+      expect(emailField.hasError, isTrue);
+
+      await tester.enterText(emailFieldFinder, '123noemail');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Not real email, so error
+      expect(emailField.hasError, isTrue);
+
+      await tester.enterText(emailFieldFinder, email);
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Real email, so no error
+      expect(emailField.hasError, isFalse);
+    });
+
+    testWidgets('Validates the password (not so strictly)', (WidgetTester tester) async {
+      await pumpMaterial(tester, const LoginPage());
+
+      final LoginFormState formState = tester.state(find.byType(LoginForm));
+
+      final FormFieldState passwordField = tester.state(passwordFieldFinder);
+
+      // Not validated yet, so no error
+      expect(passwordField.hasError, isFalse);
+
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Empty, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, 'abcdefg');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      expect(passwordField.hasError, isFalse);
```

Comment:
shouldn't that password be invalid since there is no uppercase letter and no special character?

Reply:
We don't validate the login page as strictly. It does not matter to us if the password is not following our rules... In that case we can just show it is wrong anyways.

That's why the test description also mentions "(not as strictly)"